### ISSUE TITLE
Cache for function data could live in S3 metadata

### ISSUE THREAD

**Issue by benkehoe** [garnaat/kappa#40]
If S3 is being used for the code, the cache contents related to the function code could live in the metadata on the S3 object. This would reduce the amount of state maintained locally outside of version control.

**Comment by garnaat** [comment]
True.  I would love to eliminate the local cache completely.  My only concern is that I doubt that most people currently upload their Lambda code to S3 so it might seem a bit odd to them.  On the other hand, all they would have to do is supply a bucket name and kappa could handle the rest.
